shell/browser/api/electron_api_net_log.h

```diff
@@ -6,9 +6,14 @@
 #define SHELL_BROWSER_API_ELECTRON_API_NET_LOG_H_
 
 #include "base/callback.h"
+#include "base/files/file_path.h"
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
 #include "base/values.h"
 #include "gin/handle.h"
 #include "gin/wrappable.h"
+#include "mojo/public/cpp/bindings/remote.h"
```

Comment:
I see the headers duplicated between `.cc` and `.h` file, can you keep newly added headers minimal in the `.h` file.

Reply:
I can make that change, but is this a situation where Electron is choosing to diverge from the Chromium style guide? The Google C++ Style Guide seems to suggest that the headers should be duplicated, if they're used in the `.cc` file:

> [This also applies to related headers - foo.cc should include bar.h if it uses a symbol from it even if foo.h includes bar.h.](https://google.github.io/styleguide/cppguide.html#Include_What_You_Use)

Reply:
Electron follows chromium style guide, chromium style guide https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md#forward-declarations-vs_includes has some differences to google style guide which are intentional deviations.

Reply:
Sorry, not sure I follow what you'd like with this comment, my reading of your link is to allow (and favor) forward declarations, but I don't think that deviates from the part I linked about IWYU and duplicating headers between `*.cc` and `*.h`. Which of these (or none?) are you requesting I change?

1. Remove the duplicates from `*.cc`
2. Remove the duplicates from `*.h`
3. Use forward declares in `*.h` instead

Reply:
2) Remove the duplicates from *.h, for

`This can reduce compile times and result in fewer files needing recompilation when a header changes.`

Also the change requested is purely based on what is being followed in other source files in the repo.

Reply:
Given that these types are referenced in this header, these .h files must already be in the include chain, so adding these is a no-op performance wise, and is good behavior because it reduces the likelihood that this file will start breaking in the future based on a change in #includes in unrelated files. I think it's correct to add these.